src/SimpleWebRequest.ts

```diff
@@ -236,6 +244,33 @@ export abstract class SimpleWebRequestBase<TOptions extends WebRequestOptions =
         }
     }
 
+    private static _scheduleHungRequestCleanupIfNeeded() {
+        // Schedule a cleanup timer if needed
+        if (executingList.length > 0 && hungRequestCleanupTimer === undefined) {
+            hungRequestCleanupTimer = SimpleWebRequestOptions.setTimeout(this._hungRequestCleanupTimerCallback,
+                SimpleWebRequestOptions.HungRequestCleanupIntervaMs);
+        } else if (executingList.length === 0 && hungRequestCleanupTimer) {
+            SimpleWebRequestOptions.clearTimeout(hungRequestCleanupTimer);
+            hungRequestCleanupTimer = undefined;
+        }
+    }
+
+    private static _hungRequestCleanupTimerCallback = () => {
+        hungRequestCleanupTimer = undefined;
+        executingList.filter(request => {
+            if (request._xhr && request._xhr.readyState === 4) {
+                console.warn('SimpleWebRequests found a completed XHR that hasn\'t invoked it\'s callback functions, manually responding');
```

Comment:
I wonder if we should use a callback instead of (or in addition to) this `console.warn`. For example, given the filtered list so the app can send telemetry to detect and eventually solve this problem.

Reply:
Based on what we've seen this likely has nothing to do with specific apps, but is likely a browser bug, I don't think a callback would be overly useful in that case

Reply:
Yeah, makes sense.